File: Week 1/Credit/credit.c
#include <cs50.h>
#include <stdio.h>

void checksum(long num);

int main(void)
{
    // Prompt for input
    long number = get_long("Number: ");

    // Calculate checksum
    int checksum = 0;
    for (long i = number; i > 0; i /= 100)
    {
        // Get last 2 digit each iterate
        int j = i % 100;

        // Get the tens digit
        int tens_digit = 2 * (j / 10);
        if (tens_digit >= 10)
        {
            tens_digit = (tens_digit / 10) + (tens_digit % 10);
        }
        // Get the ones digit
        int ones_digit = j % 10;

        // Sum
        checksum += tens_digit + ones_digit;
    }
    checksum %= 10;

    // Check for starting digits
    int amex_start = number / 10000000000000;
    int mastercard_start = number / 100000000000000;
    int visa16_start = number / 1000000000000000;
    int visa13_start = number / 1000000000000;

    // Print result
    if (checksum == 0)
    {
        if (amex_start == 34 || amex_start == 37)
        {
            printf("AMEX\n");
        }
        else if (mastercard_start >= 51 && mastercard_start <= 55)
        {
            printf("MASTERCARD\n");
        }
        else if (visa13_start == 4 || visa16_start == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Week 6/Credit/credit.py
# Import library
from cs50 import get_int

# Prompt user for number
number = get_int("Number: ")
number = str(number)

# Initialize counter
i = len(number) - 1
sum = 0

# Calculate checksum
while (i >= 0):
    # Calculate second-to-last digit
    if (i - 1 >= 0):
        num = int(number[i - 1]) * 2
        if (num >= 10):
            # Split digit
            num1 = num // 10
            num2 = num % 10
            # Sum those products’ digits
            sum += num1
            sum += num2
        else:
            sum += num
    # Calculate other than second-to-last digit
    sum += int(number[i])
    # Update counter
    i -= 2
checksum = sum % 10

# Compute result
if (checksum == 0):
    # AMEX
    if len(number) == 15 and (number[:2] == "34" or number[:2] == "37"):
        print("AMEX")
    # MASTERCARD
    elif len(number) == 16 and (int(number[:2]) >= 51 and int(number[:2]) <= 55):
        print("MASTERCARD")
    # VISA
    elif (len(number) == 13 or len(number) == 16) and number[0] == "4":
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")
